docs: Method of converting from SqlLite to PostgreSQL

<!--- Provide a general summary of the feature request in the Title above -->

## Feature Suggestion

<!--- If you're looking for help, please see https://backstage.io/ --->
<!--- Tell us how we could improve your experience -->

The documentation doesn't currently cover the steps needed to move cleanly from a newly templated Backstage app that was set to use SqlLite, to then convert it to using PostgreSQL.  As a pattern, it may be likely that a user:

* Templates a new app using `npx @backstage/create-app`
* Gets started by only selecting SqlLite
* (time passes... one week later) Wants to get a bit more advanced and now move their app to PostgreSQL
* After modifying `app-config.yaml` appropriately, getting a working available PostgreSQL system, the backend startup fails with a Node error about the pg library among others being missed

When you actually template a new app with PostgreSQL, you seem to also get the `pg` and `pg-connection-string` libraries listed in your `packages/backend/package.json` file.

## Possible Implementation

<!--- Not obligatory, but ideas as to the implementation of the addition or change -->

Ideally, there should be a quick paragraph that high-levels the steps necessary to move an already Backstage templated setting from SqlLite to PostgreSQL, as this is likely to be a common path.  Obviously, the nuances of every database can't be outlined, but even if it was just 2-3 steps like "run yarn add pg" then change your settings in app-config.yaml, that sort of thing.

## Context

<!--- What are you trying to accomplish? -->
<!--- Providing context (e.g. links to configuration settings, stack trace or log data) -->
<!--- helps us come up with a solution that is most useful in the real world -->

My colleague @tom-halpin used the path above, and found some documentation on one of the Roadie repos for how to perform the `app-config.yaml` changes, and he did all of the Docker work himself, but got stuck when the backend failed throwing npm errors.

In the cloned repo for example, I see these in `packages/backend/package.json`:

```diff
 "express": "^4.17.1",
     "express-promise-router": "^3.0.3",
     "knex": "^0.95.1",
+    "pg": "^8.3.0",
+    "pg-connection-string": "^2.3.0",
     "sqlite3": "^5.0.0",
     "winston": "^3.2.1"
```

So maybe even the instructions should say remove SqlLite and add PG, or something like that - a clean way of doing it sort of. (or even just a pattern that says you only need these libraries for this, or these libraries for that, something like that)

@adamdmharvey the relevant section from the logs as discussed

[1] Knex: run
[1] $ npm install pg --save
[1] Cannot find module 'pg'
[1] Require stack:
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/dialects/postgres/index.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/internal/config-resolver.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/Knex.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/index.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/knex.js
[1] - /home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js
[1] - /home/tomh/backstage-dxc/packages/backend/dist/main.js
[1] Error: Cannot find module 'pg'
[1] Require stack:
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/dialects/postgres/index.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/internal/config-resolver.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/Knex.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/index.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/knex.js
[1] - /home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js
[1] - /home/tomh/backstage-dxc/packages/backend/dist/main.js
[1]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
[1]     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
[1]     at Module.require (internal/modules/cjs/loader.js:952:19)
[1]     at require (internal/modules/cjs/helpers.js:88:18)
[1]     at Client_PG._driver (/home/tomh/backstage-dxc/node_modules/knex/lib/dialects/postgres/index.js:48:12)
[1]     at Client_PG.initializeDriver (/home/tomh/backstage-dxc/node_modules/knex/lib/client.js:178:26)
[1]     at new Client (/home/tomh/backstage-dxc/node_modules/knex/lib/client.js:71:12)
[1]     at new Client_PG (/home/tomh/backstage-dxc/node_modules/knex/lib/dialects/postgres/index.js:18:5)
[1]     at Object.knex [as default] (/home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/Knex.js:12:28)
[1]     at createPgDatabaseClient (/home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js:79:51)
[1]     at ensurePgDatabaseExists (/home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js:109:17)
[1]     at ensureDatabaseExists (/home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js:156:12)
[1]     at SingleConnectionDatabaseManager.ensureDatabase (/home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js:192:11)
[1]     at SingleConnectionDatabaseManager.getDatabase (/home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js:180:16)
[1]     at Object.getClient (/home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js:172:22)
[1]     at CatalogBuilder.build (/home/tomh/backstage-dxc/node_modules/@backstage/plugin-catalog-backend/dist/index.cjs.js:3071:68)
[1] Backend failed to start up, Error: Knex: run
[1] $ npm install pg --save
[1] Cannot find module 'pg'
[1] Require stack:
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/dialects/postgres/index.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/internal/config-resolver.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/knex-builder/Knex.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/lib/index.js
[1] - /home/tomh/backstage-dxc/node_modules/knex/knex.js
[1] - /home/tomh/backstage-dxc/node_modules/@backstage/backend-common/dist/index.cjs.js
[1] - /home/tomh/backstage-dxc/packages/backend/dist/main.js
[0] ℹ ｢wdm｣: wait until bundle finished: /

--- a/docs/tutorials/switching-sqlite-postgres.md
+++ b/docs/tutorials/switching-sqlite-postgres.md
@@ -1,0 +1,61 @@
+---
+id: switching-sqlite-postgres
+title: Switching Backstage from SQLite to PostgreSQL
+description:
+  How to get ready for deploying Backstage to production with PostgreSQL
+---
+
+The default `@backstage/create-app` database is SQLite, an in-memory database
+that's perfect for initial experimentation as it requires no environment setup.
+
+Once you're ready to deploy Backstage in production, or to have a more
+persistent development setup, you can switch the Backstage database to
+PostgreSQL.
+
+Backstage uses the [Knex](http://knexjs.org/) library, making it fairly easy to
+switch between database backends.
+
+## Install PostgreSQL
+
+First, swap out SQLite for PostgreSQL in your `backend` package:
+
+```shell
+cd packages/backend
+yarn remove sqlite3
+yarn add pg
+```
+
+## Add PostgreSQL configuration
+
+Next, modify `app-config.yaml` in the root folder to add PostgreSQL
+configuration for the backend:
+
+```diff
+backend:
+  database:
+-    client: sqlite3
+-    connection: ':memory:'
++    # config options: https://node-postgres.com/api/client
++    client: pg
++    connection:
++      host:
++        $env: POSTGRES_HOST
++      port:
++        $env: POSTGRES_PORT
++      user:
++        $env: POSTGRES_USER
++      password:
++        $env: POSTGRES_PASSWORD
++      # https://node-postgres.com/features/ssl
++      #ssl: require # see https://www.postgresql.org/docs/current/libpq-ssl.html Table 33.1. SSL Mode Descriptions (e.g. require)
++        #ca: # if you have a CA file and want to verify it you can uncomment this section
++        #$file: <file-path>/ca/server.crt
+
+```
+
+If you have a `app-config.local.yaml` for local development, a similar update
+should be made there. You can set the `POSTGRES_` environment variables prior to
+launching Backstage, or remove the $env keys and simply set values directly for
+development.
+
+The Backstage App is now ready to start up with a PostgreSQL backing database.

--- a/microsite/sidebars.json
+++ b/microsite/sidebars.json
@@ -193,7 +193,8 @@
     "Tutorials": [
       "tutorials/journey",
       "tutorials/quickstart-app-auth",
-      "tutorials/quickstart-app-plugin"
+      "tutorials/quickstart-app-plugin",
+      "tutorials/switching-sqlite-postgres"
     ],
     "Architecture Decision Records (ADRs)": [
       "architecture-decisions/adrs-overview",

